src/dataflow-types/src/client.rs

```diff
@@ -1160,14 +1337,22 @@ pub mod tcp {
         /// Connects a remote client to the specified remote dataflow server.
         pub async fn connect(addr: String) -> Result<TcpClient<T>, anyhow::Error> {
             let connection = framed_client(TcpStream::connect(&addr).await?);
-            Ok(Self { connection, addr })
+            Ok(Self {
+                errored: false,
+                connection,
+                addr,
+            })
         }
 
+        /// Reconnects the underlying `connection`.
         pub async fn reconnect(&mut self) {
             let mut connection = TcpStream::connect(&self.addr).await;
             while connection.is_err() {
+                tracing::warn!("Connect error; reconnecting");
+                std::thread::sleep(std::time::Duration::from_millis(1000));
```

Comment:
This needs to be `std::task::sleep` to avoid wedging up the coordinator. I'm going to push up a fix for that and then merge, I think!

Reply:
I decided it would be antisocial to merge, so I just pushed up the fix for you to review @frankmcsherry!